base/some.jl

```diff
@@ -0,0 +1,68 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+"""
+    Some{T}
+
+A wrapper type used with `Union{Some{T}, Void}` to distinguish between the absence
+of a value ([`nothing`](@ref)) and the presence of a `nothing` value (i.e. `Some(nothing)`).
+It can also be used to force users of a function argument or of an object field
+to explicitly handle the possibility of a field or argument being `nothing`,
+either manually or by calling [`get`](@ref) before actually using the wrapped
+value.
+"""
+struct Some{T}
+    value::T
+end
+
+eltype(::Type{Some{T}}) where {T} = T
+
+promote_rule(::Type{Some{S}}, ::Type{Some{T}}) where {S,T} = Some{promote_type(S, T)}
+promote_rule(::Type{Some{T}}, ::Type{Void}) where {T} = Union{Some{T}, Void}
+
+convert(::Type{Some{T}}, x::Some) where {T} = Some{T}(convert(T, x.value))
+convert(::Type{Union{Some{T}, Void}}, x::Some) where {T} = convert(Some{T}, x)
```

Comment:
Not completely, as `Nullable` currently supports more conversions, e.g. `convert(Nullable{Int}, nothing) === Nullable{Int}()`, which introduces an ambiguity about whether `nothing` is a valid value or a missing value. IIRC this method is needed so that you can assign e.g. `Some(1)` to a `Union{Some{Real}, Void}` field and get the automatic conversion.